src/main/java/frameworksdrivers/Gateways/StudentDbGateway.java

```diff
@@ -96,47 +99,20 @@ public boolean checkPassword(LoginStudentRequestModel requestModel) {
         return false;
     }
 
-    /**
-     * Method saves the user info stored in "student" to the database.
-     *
-     * @param student model that stores what will be saved onto the database.
-     */
-    @Override
-    public void saveUser(CreateStudentDsModel student) {
-        try{
-            Map<String, Float> courses = student.getCourseList();
-            if (courses.isEmpty()){
-               saveUserNoCourses(student);
-            } else {
-                for (Map.Entry<String, Float> course : courses.entrySet()) {
-                    String SQL = "INSERT INTO " + DATABASE_NAME +
-                            " (StudentID, Password, CourseID, CourseGrade) VALUES (?, ?, ?, ?)";
-                    PreparedStatement statement = connection.prepareStatement(SQL);
-                    statement.setString(1, student.getUsername());
-                    statement.setString(2, student.getPassword());
-                    statement.setString(3, course.getKey());
-                    statement.setFloat(4, 0.0f);
-                    statement.executeUpdate();}
-            }
-        } catch (SQLException e) {
-            System.out.println("Error with database!");
-            e.printStackTrace();
-        }
-    }
```

Comment:
This method is "deleted" and replaced with saveUserNoCourses, because now to create a student, it gets saved with a DEFAULT, which means all students will always have this DEFAULT course with no grade.

This will not cause a bug in the course database because this course doesn't exist, and so it's not an existing course id in the course database.

The student's rows with DEFAULT will never need to be updated. However it also allows for future extension by making the DEFAULT the cumulative gpa (thus upon creation all students have a CGPA of 0.0) when a course is evaluated, the DEFAULT CGPA would be updated.

Reply:
So default in this case acts as a constant that's to be saved ?